ngx_http_auth_ldap_module.c

```diff
@@ -749,6 +750,33 @@ ngx_http_auth_ldap_parse_referral(ngx_conf_t *cf, ngx_http_auth_ldap_server_t *s
     return NGX_CONF_ERROR;
 }
 
+/**
+ * Escape value for LDAP filters
+ */
+static ngx_str_t
+ngx_http_auth_ldap_escape_filter(ngx_http_request_t *r, ngx_str_t *in)
+{
+    struct berval userbv, euserbv;
+    ngx_str_t out = ngx_null_string;
+
+    if (in == NULL || in->len == 0) {
+        return out;
+    }
+
+    userbv.bv_val = in->data;
+    userbv.bv_len = in->len;
+    if (ldap_bv2escaped_filter_value(&userbv, &euserbv) != 0) {
+        return out;
+    }
+
+    out.len = euserbv.bv_len;
+    out.data = ngx_pcalloc(r->pool, out.len);
```

Comment:
Can `ngx_pcalloc` not fail?

```suggestion
    out.data = ngx_pcalloc(r->pool, out.len);
    if (!out.data) {
       return out;
    }
```

Reply:
returning `out` is not correct, because `out.len` will be set to new length but `out.data` will be null.

`return ngx_null_string` is probably better

Reply:
but tbh: Whole upstream code is not checking return code of ngx_pcalloc, so meh.

Reply:
Yeah. I saw that I will probably not use this module at all and use https://github.com/sto/ngx_http_auth_pam_module which seems much simpler and cleaner code.